block.go

```diff
@@ -65,32 +70,42 @@ func BlockParentArrayRules() serializer.ArrayRules {
 	return blockParentArrayRules
 }
 
+// EmptyBlockID returns an empty BlockID.
+func EmptyBlockID() BlockID {
+	return emptyBlockID
+}
+
 // BlockID is the ID of a Block.
-type BlockID = [BlockIDLength]byte
+type BlockID [BlockIDLength]byte
 
-// BlockIDs are IDs of blocks.
-type BlockIDs = []BlockID
+func (b BlockID) MarshalBinary() (data []byte, err error) {
+	// copy
+	return b[:], nil
+}
 
-// BlockIDFromHexString converts the given block IDs from their hex to BlockID representation.
+// ToHex converts the given block ID to their hex representation.
+func (b *BlockID) ToHex() string {
```

Comment:
why pointer method?

Reply:
See above, it avoids doing a copy of the array.